### ISSUE TITLE
Missing `hint` severity for diagnostics

### ISSUE THREAD

**Issue by simonmandlik** [DetachHead/basedpyright#840]
I'm using `basedpyright` together with `neovim`, planning to migrate from `pyright`. The only thing keeping me is the missing `hint` diagnostic severity. `Hint` severity is defined both in the [MS specification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#diagnosticSeverity) and e.g. in [neovim docs](https://neovim.io/doc/user/diagnostic.html#diagnostic-quickstart).

Is there any reason why it cannot be used in `basedpyright`?

**Comment by DetachHead** [comment]
basedpyright supports the deprecated and unnecessary diagnostic tags for some rules, just like pyright. i'm not sure what the hint diagnostic looks like but i don't think pyright uses it either. i'll look into adding support for it though

**Comment by DetachHead** [comment]
oh that's just because the default `typeCheckingMode` is different in basedpyright. see [here](https://docs.basedpyright.com/latest/benefits-over-pyright/better-defaults/#typecheckingmode) and [here](https://docs.basedpyright.com/latest/configuration/config-files/#recommended-and-all) for more info

if you want the same defaults as pyright, set `typeCheckingMode` to `"basic"` or `"standard"`:
```toml
[tool.basedpyright]
typeCheckingMode = "standard"
```

**Comment by simonmandlik** [comment]
Ok, should have read the docs thoroughly once more. For anybody's reference, setting `reportUnreachable = "unreachable"` in the config also does what I would want for this special case.

The rest of my confusion comes from my lack of understanding of how diagnostic severities and tags from basedpyright are mapped to the four possible severities in neovim. But I can find my work around

Thanks for your time, feel free to close :)

**Comment by DetachHead** [comment]
no problem, your issue gave me the idea to simplify these hint diagnostics which will hopefully reduce some of the confusion people have had about them - see #845

to summarize, all diagnostics will  support the same `"hint"` category, and for rules where it makes sense to do so, the appropriate diagnostic tag (unnecessary or deprecated) will be applied automatically when the rule is set to `"hint"`:

```toml
[tool.basedpyright]
reportUnreachable = "hint"
```

the `"unreachable"` category will still work though for backwards compatibility, but it will be deprecated and probably removed at some point in the future.
